storefact/_urls.py

```diff
@@ -44,6 +44,12 @@ def url2dict(url, raise_on_extra_params=False):
     if u'create_if_missing' in parsed['query']:
         create_if_missing = parsed['query'].pop(u'create_if_missing')[-1]  # use last appearance of key
         params['create_if_missing'] = create_if_missing in TRUEVALUES
+    if u'blob_endpoint' in parsed['query']:
```

Comment:
This looks like it can easily be buggy. I would suggest to refactor this out into a function with a good name (what is this even doing?) and add some unit tests of this function with some edge cases and some cases to show what this is expected to do.

Reply:
So, url2dict cound be something like:
```
def url2dict(url, raise_on_extra_params=False):
    ...
    params['create_if_missing'] = extract_create_if_missing(parsed)
    params['default_endpoints_protocol'] = extract_default_protocol(parsed)
    ...
```
and all of these extract_ functions can be unit tested properly (in the best case also without needing to mack anything and not needing to setup stuff in the test which is not needed for this function, e.g. create_if_missing is not needed in "extract_default_protocol").

Reply:
Makes sense. Let me implement it.

Reply:
Implemented review comments. Please review again.

Reply:
Great, lgtm! Good to go!